[BUG] CI is broken: AssertionError with message role in test_action_step_to_messages

**Describe the bug**
CI is broken: https://github.com/huggingface/smolagents/actions/runs/13647944821/job/38150144811?pr=861
```
FAILED tests/test_memory.py::test_action_step_to_messages - AssertionError: assert <MessageRole....: 'tool-call'> == <MessageRole....: 'assistant'>
  
  - assistant
  + tool-call
```

I think the CI was broken with the merge of:
- #779

--- a/tests/test_memory.py
+++ b/tests/test_memory.py
@@ -70,7 +70,7 @@ def test_action_step_to_messages():
         assert "type" in content
         assert "text" in content
     message = messages[1]
-    assert message["role"] == MessageRole.ASSISTANT
+    assert message["role"] == MessageRole.TOOL_CALL
 
     assert len(message["content"]) == 1
     text_content = message["content"][0]

